src/SignalR/clients/csharp/Client/test/UnitTests/HttpConnectionTests.Negotiate.cs

```diff
@@ -156,6 +163,84 @@ public async Task NegotiateCanHaveNewFields()
                 Assert.Equal("0rge0d00-0040-0030-0r00-000q00r00e00", connectionId);
             }
 
+            [Fact]
+            public async Task ConnectionIdGetsSetWithNegotiateProtocolGreaterThanZero()
+            {
+                string connectionId = null;
+
+                var testHttpHandler = new TestHttpMessageHandler(autoNegotiate: false);
+                testHttpHandler.OnNegotiate((request, cancellationToken) => ResponseUtils.CreateResponse(HttpStatusCode.OK,
+                    JsonConvert.SerializeObject(new
+                    {
+                        connectionId = "0rge0d00-0040-0030-0r00-000q00r00e00",
+                        negotiateVersion = 1,
+                        connectionToken = "different-id",
+                        availableTransports = new object[]
+                        {
+                            new
+                            {
+                                transport = "LongPolling",
+                                transferFormats = new[] { "Text" }
+                            },
+                        },
+                        newField = "ignore this",
+                    })));
+                testHttpHandler.OnLongPoll(cancellationToken => ResponseUtils.CreateResponse(HttpStatusCode.NoContent));
+                testHttpHandler.OnLongPollDelete((token) => ResponseUtils.CreateResponse(HttpStatusCode.Accepted));
+
+                using (var noErrorScope = new VerifyNoErrorsScope())
+                {
+                    await WithConnectionAsync(
+                        CreateConnection(testHttpHandler, loggerFactory: noErrorScope.LoggerFactory),
+                        async (connection) =>
+                        {
+                            await connection.StartAsync().OrTimeout();
+                            connectionId = connection.ConnectionId;
+                        });
+                }
+
+                Assert.Equal("0rge0d00-0040-0030-0r00-000q00r00e00", connectionId);
+            }
+
+            [Fact]
+            public async Task ConnectionTokenFieldIsIgnoredForNegotiateIdLessThanOne()
```

Comment:
What is the difference between these two tests, besides the version being different?

Reply:
This made more sense in a previous iteration when the negotiate version would change what field from the negotiate response would be exposed in the client's ConnectionId  field.